src/api/app/controllers/webui/watchlist_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Webui::WatchlistController < Webui::WebuiController
+   before_action :set_project, only: [:toggle_watch]
+
+  def toggle_watch
+    User.current.watches?(@project.name) ? unwatch_item : watch_item
+
+    if request.env['HTTP_REFERER']
+      redirect_back(fallback_location: root_path)
+    else
+      redirect_to action: project_show_path, project: @project
+    end
```

Comment:
I know you only move this but I don't think this makes sense. `redirect_back` already takes the `Referer` header into account.